app/routes/search/index.ts

```diff
@@ -1,34 +1,102 @@
+    attributes: {
+      id: 'prevOption'
+    }
+  }
+  
+  const next = totalPages <= 1 || totalPages === requestedPage ? null : {
+    href: `${path}?searchString=${encodedSearchString}&page=${requestedPage + 1}`,
+    labelText: `${requestedPage + 1} of ${totalPages} `,
+    attributes: {
+      id: 'nextOption'
+    }
+  }
+
+  return { previous, next }
+}
+
+const performSearch = (searchQuery: string, requestedPage: number) => {
+  // Get results from api and provice limit and offset as parameters
+  // expect results <= limit from offset, and totalResults
+  // {results: [], total: 20}
+
+  const offset = (requestedPage - 1) * config.search.limit
+  const { results, total } = search(searchQuery, offset)
+
+  return {
+    matches: results,
+    total: total
+  }
+}
+
+const createModel = (payload: any) => {
+  const searchQuery = decodeURIComponent(payload.searchString)
+  const requestedPage = payload.page
+  
+  const { matches, total } = performSearch(searchQuery, requestedPage)
+
+  return {
+    searchQuery,
+    ...getPaginationAttributes(total, requestedPage, '/search', searchQuery),
+    results: matches,
+    total
+  }
+}
+
 module.exports = [
-    {
-        path: '/search',
-        method: 'GET',
-        options: {
-            auth: false,
-            handler: (_request: Request, h: ResponseToolkit): ResponseObject => {
-                return h.view('search/index')
-            }
+  {
+    path: '/search',
+    method: 'GET',
+    options: {
+      auth: false,
+      validate: {
+        query: Joi.object({
+          searchString: Joi.string().optional(),
+          page: Joi.number().default(1)
+        }),
+        failAction: async (request: Request, h: ResponseToolkit, error: any) => {
+          // Todo: Add error handling to search page
+          return h.view('search/index', { ...(request.payload as Object), errorMessage: { text: error.details[0].message } }).code(400).takeover()
         }
-    },
-    {
-        path: '/search',
-        method: 'POST',
-        options: {
-            auth: false,
-            validate: {
-                payload: Joi.object({
-                    searchString: Joi.string()
-                }),
-                failAction: async (request: Request, h: ResponseToolkit, error: any) => {
-                    // Todo: Add error handling to search page
-                    return h.view('search/index', { ...(request.payload as Object), errorMessage: { text: error.details[0].message } }).code(400).takeover()
-                }
-            },
-            handler: (_request: Request, h: ResponseToolkit): ResponseObject => {
-                return h.view('search/index')
-            }
+      },
+      handler: (request: Request, h: ResponseToolkit): ResponseObject => {
+        if(!request.query.searchString) {
+          return h.view('search/index')
+        }
+
+        return h.view('search/results', createModel(request.query))
+      }
+    }
+  },
+  {
+    path: '/search',
```

Comment:
why do we need 2 search api paths with GET and POST. we are only reading data why POST required?

Reply:
So having looked at this we do need both endpoint.
The GET endpoint is required for 

- when you come to the search page for the first time
- when you click on next on pagination. This contains links with query string and get request uses query object to appropriately send results for the requested page.

The POST endpoint is required for:
- When a user searches though the search input box on the search page and on the results page. As this uses form, the request receives payload as opposed to the query object on the GET endpoint.

Also both endpoint behave differently when no searchString is present. The GET endpoint responds back with the search index page. The POST endpoint responds back with an error on the index page.

So overall we do need both endpoints as they are slightly doing different things and responding to different kind of request.

Reply:
Even though it is using the differently(querystring/payload) both are doing the read operation from the server. Post should only used for write operation for example creating the payment data in this case as per rest standard. to change the behaviour we can return diff result based on the bask-end condition.

Reply:
as dicussed with Sid , this is approved in MPDP-228